cli/file_fetcher.rs

```diff
@@ -1758,4 +1803,40 @@ mod tests {
       .to_owned();
     assert_eq!(filter_shebang(code), "\nconsole.log('hello');\n".as_bytes());
   }
+
+  #[test]
+  fn test_fetch_with_etag() {
+    let http_server_guard = crate::test_util::http_server();
+    let (_temp_dir, fetcher) = test_setup();
+    let module_url =
+      Url::parse("http://127.0.0.1:4545/etag_script.ts").unwrap();
+
+    let fut = async move {
+      let source = fetcher
+        .fetch_remote_source_async(&module_url, false, false, 1)
+        .await;
+      assert!(source.is_ok());
+      let source = source.unwrap();
+      assert_eq!(source.source_code, b"console.log('etag')");
+      assert_eq!(&(source.media_type), &msg::MediaType::JavaScript);
+
+      let headers = fetcher.get_source_code_headers(&module_url);
+      assert_eq!(headers.etag, Some("33a64df551425fcc55e".to_string()));
+
+      // Forcibly change the contents of the cache file and request
+      // it again with the cache parameters turned off.
+      // If the fetched content changes, the cached content is used.
+      fetcher
```

Comment:
Great, can you present the other situation as well? Ie. you download file once and then try to download it again if cache switched off. The second download should not modify `headers` file so you can compare timestamp of `headers` file before and after second fetch